scripts/release.js

```diff
@@ -0,0 +1,189 @@
+const {execSync} = require('child_process');
+const minimist = require('minimist');
+const {prompt} = require('promptly');
+const standardVersion = require('standard-version');
+
+const args = minimist(process.argv.slice(2));
+
+const DEFAULT_UPSTREAM_REPO_BRANCH = 'master';
+const DEFAULT_UPSTREAM_REPO_ORIGIN = 'origin';
+const branch = args['repo-branch'] || DEFAULT_UPSTREAM_REPO_BRANCH;
+const origin = args['repo-origin'] || DEFAULT_UPSTREAM_REPO_ORIGIN;
+
+const distTag = args['npm-dist-tag'];
+const dryRun = args['dry-run'];
+const prerelease = args.prerelease;
+const releaseAs = args['release-as'];
+const sign = args.sign;
+
+// eslint-disable-next-line no-confusing-arrow
+const dryRunMaybe = () => dryRun ? leftPad1(chalk.yellow('(DRY RUN)')) : '';
+const execSyncInherit = (cmd) => execSync(cmd, {stdio: 'inherit'});
+// eslint-disable-next-line no-confusing-arrow
+const leftPad1 = (str) => ' ' + str;
+// eslint-disable-next-line no-confusing-arrow
+const leftPad1Maybe = (str) => str ? leftPad1(str) : str;
+const log = (mark, str) => console.log(mark, str.filter(s => !!s).join(' '));
+const logError = (...str) => log(chalk.red('✘'), str);
+const logInfo = (...str) => log(chalk.blue('ℹ'), str);
+const logSuccess = (...str) => log(chalk.green('✔'), str);
+const logWarning = (...str) => log(chalk.yellow('‼︎'), str);
+
+const yesNoValidator = (input) => {
+  const _input = input && input[0].toLowerCase();
+  if (!['y', 'n'].includes(_input)) {
+    throw new Error(chalk.red('✘') + leftPad1(`Please answer [y]es or [n]o.`));
+  }
+  return _input;
+};
+
+const promptOpts = {default: 'blank', validator: yesNoValidator};
+
+const proceedAnywayPrompt = async () => {
+  let answer = await prompt(
+    `${chalk.yellow('⁉︎')} Proceed anyway? [y/n]`,
+    promptOpts
+  );
+  if (answer === 'n') {
+    logWarning(`Stopping right here.`);
+    process.exit(0);
+  }
+};
+
+const dryRunPrompt = async () => {
+  let answer = await prompt(
+    `${chalk.blue('⁇')} This is ${chalk.yellow('NOT')} a --dry-run.` +
+      leftPad1(`Did you complete a successful --dry-run first? [y/n]`),
+    promptOpts
+  );
+  if (answer === 'n') await proceedAnywayPrompt();
+};
+
+(async () => {
+  try {
+    if (!dryRun) await dryRunPrompt();
+
+    logInfo(`Determining the current branch...`);
+
+    let currentBranch;
+    try {
+      currentBranch = execSync(`git rev-parse --abbrev-ref HEAD`)
+        .toString()
+        .trim();
+    } catch (e) {
+      logError(`Couldn't determine the branch. Please check the error above.`);
+      throw new Error();
+    }
+
+    if (currentBranch !== branch) {
+      logError(
+        `Current branch '${currentBranch}' is not the same as release branch`,
+        `'${branch}'. Please checkout the release branch before rerunning this`,
+        `script or rerun with '--repo-branch ${currentBranch}'.`
+      );
+      throw new Error();
+    }
+
+    logSuccess(`Current branch and relase branch are the same.`);
```

Comment:
relase => release

Reply:
Good catch, fixed! (and rebased with master)

Reply:
I'll squash the post-feedback commits when it's time to merge into master.

Reply:
Done.